gnomad_qc/v4/annotations/generate_freq.py

```diff
@@ -466,35 +491,54 @@ def combine_freq_hts(
     return freq_ht
 
 
-# Functions to correct frequencies and hists for high ab hets.
-def create_high_ab_age_hists_expr(ht: hl.Table, age_group_key="sample_age_bin"):
+def correct_for_high_ab_hets(ht: hl.Table, af_threshold: float = 0.01) -> hl.Table:
     """
-    Create histograms of high ab counts using age bins to account for high AB hets becoming hom alts.
+    Add documentation.
 
-    :param ht: Hail Table containing age hists, AB annotation.
-    :param age_group_key: Age group key to use for age histogram.
-    :return: Hail struct containing age histogram of high ab counts.
+    :param ht:
+    :param af_threshold:
+    :return:
     """
-    non_range_entries = hl.set(["n_larger", "n_smaller"])
-    age_bins_indices = hl.sorted(
-        hl.enumerate(ht["freq_meta"], index_first=False)
-        .filter(lambda x: x[0].contains(age_group_key))
-        .map(lambda x: (x[0][age_group_key], x[1]))
+    # TODO: Comment
+    call_stats_expr = hl.map(
+        lambda f, g: hl.struct(
+            AC=hl.int32(f.AC + g),
+            AN=f.AN,
+            homozygote_count=f.homozygote_count + g,
+            AF=hl.if_else(f.AN > 0, (f.AC + g) / f.AN, hl.missing(hl.tfloat64)),
+        ),
+        ht.freq,
+        ht.high_ab_hets_by_group,
     )
 
-    age_bins_indices_dict = hl.dict(age_bins_indices)
-    age_bin_indices_no_edges = age_bins_indices.filter(
-        lambda x: ~non_range_entries.contains(x[0])
-    )
-    return hl.struct(
-        bin_freq=hl.starmap(
-            lambda x, y: ht.high_ab_hets_by_group[y],
-            age_bin_indices_no_edges,
+    # TODO: Comment
+    qual_hist_expr = {
+        f"ab_adjusted_{x}": ht[x].annotate(
+            ab_hist_alt=ht.high_ab_het_adjusted_ab_hists[
+                f"ab_hist_alt{'' if x.startswith('raw') else '_adj'}"
+            ]
+        )
+        for x in FREQ_ROW_FIELDS
+        if "qual_hist" in x
+    }
+
+    # TODO: Comment
+    no_ab_adjusted_expr = {f"ab_adjusted_{x}": ht[x] for x in FREQ_ROW_FIELDS}
+    ht = ht.select(
+        *FREQ_ROW_FIELDS,
```

Comment:
I think it's worth keeping `high_ab_hets_by_group` in here so we can spot check the correction. We can use the uncorrected HT and corrected during testing too I guess but the finalize arg you mention below could just drop it. We could also put it back in FREQ_ROW_FIELDS even though it has high_ab in the name?
```suggestion
        "high_ab_hets_by_group",
        *FREQ_ROW_FIELDS,
```